app/models/reading.rb

```diff
@@ -2,7 +2,7 @@ class Reading < ActiveRecord::Base
   belongs_to :user
   belongs_to :work
 
-  after_save :expire_cached_home_marked_for_later, if: :toread?
+  after_save :expire_cached_home_marked_for_later, if: :toread_changed?
   after_destroy :expire_cached_home_marked_for_later, if: :toread?
```

Comment:
I don't think so -- when the reading gets destroyed, it doesn't change `toread`. If I try `toread_changed?` here, it doesn't do `expire_cached_home_marked_for_later`.